lib/matplotlib/contour.py

```diff
@@ -702,6 +702,11 @@ def __init__(self, ax, *args,
         self._extend_min = self.extend in ['min', 'both']
         self._extend_max = self.extend in ['max', 'both']
         if self.colors is not None:
+            if isinstance(self.colors, str):
+                colors = [colors]
+            else:
+                colors = colors
```

Comment:
Why don't you just normalize `self.colors`? Having both `self.colors` and `colors` with potentially different content is confusing and may lead to bugs later. Also, the `else` clause does nothing?!?

```suggestion
            if isinstance(self.colors, str):
                self.colors = [self.colors]
```

Reply:
I left `self.colors` as is because it's public so a user might access it expecting to find whatever they passed in.

> Also, the else clause does nothing?!?

Ooops.  I wrote these as `= [self.colors]` and `=self.colors` and must have been too thorough with the find/replace.  I also forgot that I'm still in `__init__` so have `colors` lying around.

Reply:
> I left `self.colors` as is because it's public so a user might access it expecting to find whatever they passed in.

Ok. AFAIK we don't have a clear policy on that. Personally, I think that we can/should do input normalization. I consider a single color a convenience input API, but not a valid core state of the Artist. But that can be discussed separately. Your PR keeps the current behavior unchanged. 👍

Reply:
Off topic, but I don't think we should be worried about keeping track of the exact form of the users arguments for them. The caveat would be if self.colors turned into something that was not easily queried, which I don't think is the case.

Reply:
Arguably this _shouldn't_ be a public attribute because changing it after instantiation has no effect.

Reply:
You are right. AFAICS we never use it outside of `__init__`. Therefore, we can simply deprecate it.